### PR TITLE
Fix Start/Stop/Clean All. Closes #55. Progress: #47, #54

### LINKED ISSUE #55: template_parser.py should check for running containers before parsing template
Checking for running containers will prevent duplicate containers from being prepped for vent-management.

### PR DESCRIPTION
**Pull request by Joecakes4u**

### CODE REVIEW

File: template_parser.py [hunk 1]
```diff
@@ -330,25 +366,43 @@ def read_template_types(template_type, container_cmd, template_dir, plugins_dir)
...
         elif template_execution == "clean":
-            os.system("docker ps -a | grep "+template_type+" | awk '{print $1}' | xargs docker kill")
-            os.system("docker ps -a | grep "+template_type+" | awk '{print $1}' | xargs docker rm")
+            if template_type == "all":
+                for x in ["visualization", "core", "active", "passive"]:
+                    os.system("docker ps -aqf name=\""+x+"\" | xargs docker kill")
+                    os.system("docker ps -aqf name=\""+x+"\" | xargs docker rm")
+            else:
+                os.system("docker ps -a | grep "+template_type+" | awk '{print $1}' | xargs docker kill")
+                os.system("docker ps -a | grep "+template_type+" | awk '{print $1}' | xargs docker rm")
```
Comment (cglewis) [inline template_parser.py]: shouldn't these be changed to the other style from `394-395` as well?

File: template_parser.py [hunk 2]
```diff
@@ -330,25 +366,43 @@ def read_template_types(template_type, container_cmd, template_dir, plugins_dir)
...
+            pass
     except Exception as e:
+        print e
         pass
     return info_name, service_schedule, tool_core, tool_dict, delay_sections
 
 def main():
-    template_dir = "/var/lib/docker/data/templates/"
-    plugins_dir = "/var/lib/docker/data/plugins/"
+    path_dirs = PathDirs()
```
Comment (cglewis) [inline template_parser.py]: this should be created under `__main__` and then passed in as an argument to `main()`

File: template_parser.py [hunk 3]
```diff
@@ -330,25 +366,43 @@ def read_template_types(template_type, container_cmd, template_dir, plugins_dir)
                                 instructions['Volumes'] = {"/"+section+"-data": {}}
                                 tool_dict[template_type+"-"+section] = instructions
         else:
-            info_name = "\"all\""
+            pass
     except Exception as e:
+        print e
```
Comment (cglewis) [inline template_parser.py]: logging should probably be removed

File: template_parser.py [hunk 4]
```diff
@@ -89,6 +111,15 @@ def read_template_types(template_type, container_cmd, template_dir, plugins_dir)
...
             except Exception as e:
                 sections = []
+            running_containers = subprocess.check_output("docker ps | awk \"{print \$NF}\"", shell=True).split("\n")
+            t_sections = []
+            # remove sections that represent running containers
+            for section in sections:
+                for container in running_containers:
+                    if section in container and template_type in container:
+                        t_sections.append(section)
+                        print section
```
Comment (cglewis) [inline template_parser.py]: logging should probably be removed

File: template_parser.py [hunk 5]
```diff
@@ -89,6 +111,15 @@ def read_template_types(template_type, container_cmd, template_dir, plugins_dir)
...
+            sections = [x for x in sections if x not in t_sections]
```
Comment (cglewis) [inline template_parser.py]: where do `sections` and `t_sections` get used?

File: template_parser.py [hunk 6]
```diff
@@ -330,25 +366,44 @@ def read_template_types(template_type, container_cmd, template_dir, plugins_dir)
...
     except Exception as e:
+        print e
         pass
     return info_name, service_schedule, tool_core, tool_dict, delay_sections
 
 def main():
-    template_dir = "/var/lib/docker/data/templates/"
-    plugins_dir = "/var/lib/docker/data/plugins/"
+    """main method for template_parser. Based on the action argument given, performs the actions on the correct template files"""
+    path_dirs = PathDirs()
```
Comment (cglewis) [inline template_parser.py]: this should be created under __main__ and then passed in as an argument to main()

File: template_parser.py (hunk 5 above)
Comment (Joecakes4u) [inline template_parser.py]: This checks for sections in a namespace's template, removing sections for containers that are already running. Is this a redundant check?

File: plugin_parser.py [hunk 7]
```diff
@@ -167,6 +165,7 @@ def add_plugins(path_dirs, plugin_url):
         os.system("/bin/sh /data/build_images.sh")
         return
     except Exception as e:
+        print e
```
Comment (cglewis) [inline plugin_parser.py]: no need for logging here

**Pull request merged**